core/src/main/scala/org/apache/spark/deploy/security/HadoopDelegationTokenProvider.scala

```diff
@@ -18,15 +18,17 @@
 package org.apache.spark.deploy.security
```

Comment:
I remembered that things under `org.apache.spark.deploy` are not really "public" in Spark's view (i.e. they're not publicly documented). So leaving this interface here would mean it still remains in an awkward situation.

We should probably move the interface to `org.apache.spark.security` (the rest could stay here). Up to you if you want to do it here or in a separate patch.

Reply:
I think it's not a heavy stuff so put it here.